src/solver.jl

```diff
@@ -37,7 +37,13 @@ function make_optimization_model(
 
     optimization_model = Model(optimizer)
     @variable(optimization_model, x[1:n])
-    @objective(optimization_model, sense, objective(model)' * x)
+    let obj = objective(model)
+        if obj isa AbstractVector
+            @objective(optimization_model, sense, obj' * x)
+        else
+            @objective(optimization_model, sense, x' * obj * [x; 1])
+        end
+    end
```

Comment:
This would cover 99% of use cases but not if someone has a weird objective like x^2 + x

Reply:
yeah actually that's covered there, the extra column is precisely for the `+x`.

Reply:
e.g. if you have 2 vars and want to do x[1]^2+x[1], you basically do
objective(m) = [1 0 1; 0 0 0]

Reply:
slightly harder to construct though?

Reply:
ah yes there's gonna be a tool for that